testing/hello_events/test_hello_events_c.cpp

```diff
@@ -344,10 +342,39 @@ class hello_events_c_fpgad_p : public ::testing::TestWithParam<std::string> {
  * @test       main2
  * @brief      Test: hello_events_main
  * @details    When a suitable accelerator device is found,<br>
- *             hello_events_main runs to completion,<br>
+ *             hello_events_main (mock) times out,<br>
+ *             and the fn returns non-zero.<br>
+ */
+TEST_P(mock_hello_events_c_fpgad_p, main2) {
+  char zero[20];
+  char one[20];
+  char two[20];
+  strcpy(zero, "hello_events");
+  strcpy(one, "-B");
+  sprintf(two, "%d", platform_.devices[0].bus);
+
+  char *argv[] = { zero, one, two };
+
+  EXPECT_NE(hello_events_main(3, argv), 0);
+}
+
+INSTANTIATE_TEST_CASE_P(mock_hello_events_c_fpgad, mock_hello_events_c_fpgad_p,
+                        ::testing::ValuesIn(test_platform::mock_platforms()));
+
+
+class hw_hello_events_c_fpgad_p : public mock_hello_events_c_fpgad_p {
+ protected:
+  hw_hello_events_c_fpgad_p() {}
+};
+
+/**
+ * @test       main2
+ * @brief      Test: hello_events_main
+ * @details    When a suitable accelerator device is found,<br>
+ *             hello_events_main (hw) runs to completion,<br>
  *             and the fn returns zero.<br>
  */
-TEST_P(hello_events_c_fpgad_p, main2) {
+TEST_P(hw_hello_events_c_fpgad_p, main2) {
```

Comment:
This test seems to be failing for MCP with FPGA_EXCEPTION - poll timeout occurred.

Reply:
Worked for me on cpt-ncf022.  Did you try running from my branch?

Reply:
This test is also failing for me on bmp012 with error registering FME event, FPGA_NO_DAEMON.

Reply:
It passes when running with sudo.

Reply:
Ah yes, sudo is required, since we are trying to open a handle to the FME.  This wouldn't be required on mock, of course, since there is no enforcement of device permissions.  Not sure how to best handle this.  @r-rojo Thought about tests that require sudo?

Reply:
Might just assume that when running on hw that the test runner has root privileges.